> I would expect an IDE solves the formatting.

I would not expect everyone to use an IDE, let alone expect those IDEs to have the same idea of good formatting :-) Anyway, this was more of a thought and should be discussed in an issue first, if we want that.

> they work great if you have test coverage, but if the code is formatted wrong, at least it's noticeable, if it looks right but does wrong and there's no test, it's hard to detect. More tests -> more belief

I don't quite understand your concern, if it is really formatted wrong then that is a syntax error, everything else is an aesthetic choice (at least black should never change the result of the code, since it checks that the resulting AST stays effectively the same).

For this PR, I'd propose this:
- add flake8-builtins to the dev dependencies and the github action.
  This finds the following issues:
```
$ flake8 mslib tests
mslib/index.py:196:5: A001 variable "help" is shadowing a Python builtin
mslib/mscolab/models.py:39:5: A003 class attribute "id" is shadowing a Python builtin
mslib/mscolab/models.py:108:5: A003 class attribute "id" is shadowing a Python builtin
mslib/mscolab/models.py:130:5: A003 class attribute "id" is shadowing a Python builtin
mslib/mscolab/models.py:161:5: A003 class attribute "id" is shadowing a Python builtin
mslib/mscolab/models.py:185:5: A003 class attribute "id" is shadowing a Python builtin
mslib/msui/mpl_pathinteractor.py:1153:9: A001 variable "map" is shadowing a Python builtin
mslib/msui/mpl_qtwidget.py:1502:5: A003 class attribute "map" is shadowing a Python builtin
mslib/msui/mscolab.py:846:17: A001 variable "type" is shadowing a Python builtin
mslib/msui/msui_mainwindow.py:112:18: A002 argument "type" is shadowing a Python builtin
mslib/msui/wms_control.py:82:16: A002 argument "format" is shadowing a Python builtin
mslib/msui/wms_control.py:1244:66: A002 argument "format" is shadowing a Python builtin
mslib/mswms/gallery_builder.py:726:9: A001 variable "id" is shadowing a Python builtin
mslib/support/qt_json_view/datatypes.py:22:5: A003 class attribute "next" is shadowing a Python builtin
mslib/support/qt_json_view/datatypes.py:165:5: A003 class attribute "next" is shadowing a Python builtin
mslib/support/qt_json_view/datatypes.py:203:5: A003 class attribute "next" is shadowing a Python builtin
mslib/utils/qt.py:283:27: A002 argument "object" is shadowing a Python builtin
tests/_test_mscolab/test_models.py:64:9: A001 variable "id" is shadowing a Python builtin
tests/_test_mscolab/test_utils.py:42:5: A003 class attribute "id" is shadowing a Python builtin
tests/_test_msui/test_mscolab_merge_waypoints.py:140:121: E501 line too long (121 > 120 characters)
tests/_test_utils/test_migration.py:138:13: A001 variable "all" is shadowing a Python builtin
tests/_test_utils/test_migration.py:157:13: A001 variable "all" is shadowing a Python builtin
tests/_test_utils/test_multidict.py:58:24: A002 argument "object" is shadowing a Python builtin
tests/_test_utils/test_multidict.py:59:9: A001 variable "dict" is shadowing a Python builtin
```
- fix those issues
- merge
- discuss ruff and code formatting in general in a different issue

--- a/.github/workflows/python-flake8.yml
+++ b/.github/workflows/python-flake8.yml
@@ -30,5 +30,5 @@ jobs:
     - name: Lint with flake8
       run: |
         python -m pip install --upgrade pip
-        pip install flake8
+        pip install flake8 flake8-builtins
         flake8 --count --max-line-length=127 --statistics mslib tests

--- a/mslib/index.py
+++ b/mslib/index.py
@@ -193,7 +193,7 @@ def create_app(name="", imprint=None, gdpr=None):
                             headers={"Content-disposition": f"attachment; filename={filename.split('-')[0]}.py"})
 
     @APP.route("/mss/help")
-    def help():
+    def help():  # noqa: A001
         _file = os.path.join(DOCS_SERVER_PATH, 'static', 'docs', 'help.md')
         html_overrides = ('<img alt="Waypoint Tutorial" '
                           'src="https://mss.readthedocs.io/en/stable/_images/tutorial_waypoints.gif" />',

--- a/mslib/mscolab/models.py
+++ b/mslib/mscolab/models.py
@@ -36,7 +36,7 @@ from mslib.mscolab.message_type import MessageType
 class User(db.Model):
 
     __tablename__ = 'users'
-    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)  # noqa: A003
     username = db.Column(db.String(255))
     emailid = db.Column(db.String(255), unique=True)
     password = db.Column(db.String(255), unique=True)
@@ -105,7 +105,7 @@ class User(db.Model):
 class Permission(db.Model):
 
     __tablename__ = 'permissions'
-    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)  # noqa: A003
     op_id = db.Column(db.Integer, db.ForeignKey('operations.id'))
     u_id = db.Column(db.Integer, db.ForeignKey('users.id'))
     access_level = db.Column(db.Enum("admin", "collaborator", "viewer", "creator", name="access_level"))
@@ -127,7 +127,7 @@ class Permission(db.Model):
 class Operation(db.Model):
 
     __tablename__ = "operations"
-    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)  # noqa: A003
     path = db.Column(db.String(255), unique=True)
     category = db.Column(db.String(255))
     description = db.Column(db.String(255))
@@ -158,7 +158,7 @@ class Operation(db.Model):
 class Message(db.Model):
 
     __tablename__ = "messages"
-    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)  # noqa: A003
     op_id = db.Column(db.Integer, db.ForeignKey('operations.id'))
     u_id = db.Column(db.Integer, db.ForeignKey('users.id'))
     text = db.Column(db.Text)
@@ -182,7 +182,7 @@ class Message(db.Model):
 class Change(db.Model):
 
     __tablename__ = "changes"
-    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)  # noqa: A003
     op_id = db.Column(db.Integer, db.ForeignKey('operations.id'))
     u_id = db.Column(db.Integer, db.ForeignKey('users.id'))
     commit_hash = db.Column(db.String(255), default=None)

--- a/mslib/msui/mpl_pathinteractor.py
+++ b/mslib/msui/mpl_pathinteractor.py
@@ -1150,8 +1150,8 @@ class HPathInteractor(PathInteractor):
         # (bounds = left, bottom, width, height)
         ax_bounds = self.plotter.ax.bbox.bounds
         diagonal = math.hypot(round(ax_bounds[2]), round(ax_bounds[3]))
-        map = self.plotter.map
-        map_delta = get_distance(map.llcrnrlat, map.llcrnrlon, map.urcrnrlat, map.urcrnrlon)
+        plot_map = self.plotter.map
+        map_delta = get_distance(plot_map.llcrnrlat, plot_map.llcrnrlon, plot_map.urcrnrlat, plot_map.urcrnrlon)
         km_per_px = map_delta / diagonal
 
         return km_per_px * px

--- a/mslib/msui/mpl_qtwidget.py
+++ b/mslib/msui/mpl_qtwidget.py
@@ -1499,7 +1499,7 @@ class MplTopViewCanvas(MplCanvas):
         self.pdlg.close()
 
     @property
-    def map(self):
+    def map(self):  # noqa: A003
         return self.plotter.map
 
     def init_map(self, model=None, **kwargs):

--- a/mslib/msui/mscolab.py
+++ b/mslib/msui/mscolab.py
@@ -843,10 +843,10 @@ class MSUIMscolab(QtCore.QObject):
                     self.add_proj_dialog.buttonBox.button(QtWidgets.QDialogButtonBox.Ok).setEnabled(False)
 
             def browse():
-                type = self.add_proj_dialog.cb_ImportType.currentText()
+                import_type = self.add_proj_dialog.cb_ImportType.currentText()
                 file_type = ["Flight track (*.ftml)"]
-                if type != 'FTML':
-                    file_type = [f"Flight track (*.{self.ui.import_plugins[type][1]})"]
+                if import_type != 'FTML':
+                    file_type = [f"Flight track (*.{self.ui.import_plugins[import_type][1]})"]
 
                 file_path = get_open_filename(
                     self.ui, "Open Flighttrack file", "", ';;'.join(file_type))
@@ -856,7 +856,7 @@ class MSUIMscolab(QtCore.QObject):
                         with open_fs(fs.path.dirname(file_path)) as file_dir:
                             file_content = file_dir.readtext(file_name)
                     else:
-                        function = self.ui.import_plugins[type][0]
+                        function = self.ui.import_plugins[import_type][0]
                         ft_name, waypoints = function(file_path)
                         model = ft.WaypointsTableModel(waypoints=waypoints)
                         xml_doc = model.get_xml_doc()

--- a/mslib/msui/msui_mainwindow.py
+++ b/mslib/msui/msui_mainwindow.py
@@ -109,7 +109,7 @@ class QFlightTrackListWidgetItem(QtWidgets.QListWidgetItem):
     """
 
     def __init__(self, flighttrack_model, parent=None,
-                 type=QtWidgets.QListWidgetItem.UserType):
+                 user_type=QtWidgets.QListWidgetItem.UserType):
         """Item class for the list widget that accommodates the open flight
            tracks.
 
@@ -123,7 +123,7 @@ class QFlightTrackListWidgetItem(QtWidgets.QListWidgetItem):
         """
         view_name = flighttrack_model.name
         super().__init__(
-            view_name, parent, type)
+            view_name, parent, user_type)
 
         self.parent = parent
         self.flighttrack_model = flighttrack_model

--- a/mslib/msui/wms_control.py
+++ b/mslib/msui/wms_control.py
@@ -79,7 +79,8 @@ class MSUIWebMapService(ogcwms.WebMapService):
     """
 
     def getmap(self, layers=None, styles=None, srs=None, bbox=None,
-               format=None, size=None, time=None, init_time=None,
+               format=None,  # noqa: A002
+               size=None, time=None, init_time=None,
                path_str=None, level=None, transparent=False, bgcolor='#FFFFFF',
                time_name="time", init_time_name="init_time",
                exceptions='XML', method='Get',
@@ -1241,7 +1242,9 @@ class WMSControlWidget(QtWidgets.QWidget, ui.Ui_WMSDockWidget):
         return os.path.join(self.wms_cache, hashlib.md5(urlstr.encode('utf-8')).hexdigest() + ending)
 
     def retrieve_image(self, layers=None, crs="EPSG:4326", bbox=None, path_string=None,
-                       width=800, height=400, transparent=False, format="image/png"):
+                       width=800, height=400, transparent=False,
+                       format="image/png",  # noqa: A002
+                       ):
         """Retrieve an image of the layer currently selected in the
            GUI elements from the current WMS provider. If caching is
            enabled, first check the cache for the requested image. If

--- a/mslib/mswms/gallery_builder.py
+++ b/mslib/mswms/gallery_builder.py
@@ -723,8 +723,8 @@ def add_image(path, plot, plot_object, generate_code=False, sphinx=False, url_pr
                 markdown = add_times(itime, [vtime], markdown)
             plot_htmls[f"{l_type}_{dataset}{plot_object.name}"] = markdown
 
-        id = img_path.split("-" + f"{level}".replace(" ", "_").replace(":", "_").replace("-", "_"))[0]
-        if not any([id in html for html in plots[l_type]]):
+        img_id = img_path.split("-" + f"{level}".replace(" ", "_").replace(":", "_").replace("-", "_"))[0]
+        if not any([img_id in html for html in plots[l_type]]):
             plots[l_type].append(image_md(
                 img_path, plot_object.name, code_path if generate_code else None,
                 f"{plot_object.title}" + (f"<br>{plot_object.abstract}"

--- a/mslib/support/qt_json_view/datatypes.py
+++ b/mslib/support/qt_json_view/datatypes.py
@@ -19,7 +19,7 @@ class DataType:
         """Logic to define whether the given data matches this type."""
         raise NotImplementedError
 
-    def next(self, model, data, parent):
+    def next(self, model, data, parent):  # noqa: A003
         """Implement if this data type has to add child items to itself."""
         pass
 
@@ -162,7 +162,7 @@ class ListType(DataType):
     def matches(self, data):
         return isinstance(data, list)
 
-    def next(self, model, data, parent):
+    def next(self, model, data, parent):  # noqa: A003
         for i, value in enumerate(data):
             type_ = match_type(value)
             key_item = self.key_item(
@@ -200,7 +200,7 @@ class DictType(DataType):
     def matches(self, data):
         return isinstance(data, dict)
 
-    def next(self, model, data, parent):
+    def next(self, model, data, parent):  # noqa: A003
         for key, value in data.items():
             type_ = match_type(value)
             key_item = self.key_item(key, datatype=type_, model=model)

--- a/mslib/utils/qt.py
+++ b/mslib/utils/qt.py
@@ -280,8 +280,8 @@ class CheckableComboBox(QtWidgets.QComboBox):
         self.updateText()
         super().resizeEvent(event)
 
-    def eventFilter(self, object, event):
-        if object == self.lineEdit():
+    def eventFilter(self, obj, event):
+        if obj == self.lineEdit():
             if event.type() == QtCore.QEvent.MouseButtonRelease:
                 if self.closeOnLineEditClick:
                     self.hidePopup()
@@ -290,7 +290,7 @@ class CheckableComboBox(QtWidgets.QComboBox):
                 return True
             return False
 
-        if object == self.view().viewport():
+        if obj == self.view().viewport():
             if event.type() == QtCore.QEvent.MouseButtonRelease:
                 index = self.view().indexAt(event.pos())
                 item = self.model().item(index.row())

--- a/requirements.d/development.txt
+++ b/requirements.d/development.txt
@@ -2,6 +2,7 @@
 # $ conda create --name <env> --file <this file>
 # platform: linux-64
 pep8
+flake8-builtins
 py
 mock
 pycodestyle
@@ -25,4 +26,3 @@ dnspython>=2.0.0, <2.3.0
 gsl==2.7.0
 boa
 xmlschema<2.5.0
-

--- a/tests/_test_mscolab/test_models.py
+++ b/tests/_test_mscolab/test_models.py
@@ -61,8 +61,8 @@ class Test_User(TestCase):
     def test_verify_auth_token(self):
         user = User(self.userdata[0], self.userdata[1], self.userdata[2])
         token = user.generate_auth_token()
-        id = user.verify_auth_token(token)
-        assert user.id == id
+        uid = user.verify_auth_token(token)
+        assert user.id == uid
 
     def test_verify_password(self):
         user = User(self.userdata[0], self.userdata[1], self.userdata[2])

--- a/tests/_test_mscolab/test_utils.py
+++ b/tests/_test_mscolab/test_utils.py
@@ -39,7 +39,7 @@ from mslib.mscolab.sockets_manager import setup_managers
 
 
 class Message:
-    id = 1
+    id = 1  # noqa: A003
     u_id = 2
 
     class user:

--- a/tests/_test_utils/test_migration.py
+++ b/tests/_test_utils/test_migration.py
@@ -135,9 +135,9 @@ class TestMigration:
             result = dict()
             result[default] = mailid
             assert config_loader_before_nine(dataset="MSS_auth") == {"https://www.your-mscolab-server.de": "youruser"}
-            all = config_loader_before_nine()
+            config = config_loader_before_nine()
             # old version knows MSCOLAB_mailid
-            assert "MSCOLAB_mailid" in all.keys()
+            assert "MSCOLAB_mailid" in config.keys()
             new_version = JsonConversion()
             # converting and storing
             new_version.change_parameters()
@@ -154,6 +154,6 @@ class TestMigration:
             # added MSCOLAB_mailid to the url based on default_MSCOLAB
             mss_auth = config_loader(dataset="MSS_auth")
             assert mss_auth == {"https://www.your-mscolab-server.de": mailid}
-            all = config_loader()
+            config = config_loader()
             # new version forgot about MSCOLAB_mailid
-            assert "MSCOLAB_mailid" not in all.keys()
+            assert "MSCOLAB_mailid" not in config.keys()

--- a/tests/_test_utils/test_multidict.py
+++ b/tests/_test_utils/test_multidict.py
@@ -55,9 +55,9 @@ class TestCIMultiDict:
                     return v
             raise KeyError(repr(key))
 
-    def test_multidict(object):
-        dict = TestCIMultiDict.CaseInsensitiveMultiDict([('title', 'MSS')])
+    def test_multidict(self):
+        test_dict = TestCIMultiDict.CaseInsensitiveMultiDict([('title', 'MSS')])
         dict_multidict = multidict.CIMultiDict([('title', 'MSS')])
         assert 'title' in dict_multidict
         assert 'tiTLE' in dict_multidict
-        assert dict_multidict['Title'] == dict['tITLE']
+        assert dict_multidict['Title'] == test_dict['tITLE']

